runtime/executor/src/main/java/edu/snu/coral/runtime/executor/data/block/FileBlock.java

```diff
@@ -120,10 +120,20 @@ private void writeSerializedPartitions(final Iterable<SerializedPartition<K>> se
         final K key = partitionMetadata.getKey();
         if (keyRange.includes(key)) {
           // The key value of this partition is in the range.
+          final long availableBefore = fileStream.available();
+          final LimitedInputStream limitedInputStream =
```

Comment:
Now it seems reasonable to me. I have one more (and maybe last) suggestion.
How about include this stream in the `StreamChainer`?

Reply:
We decided to not put it in `StreamChainer` because the size of the partition have to be known before it's construction.